docs/api_reference/src/quantities.tex

```diff
@@ -109,158 +624,5492 @@
+to \tcode{quantity_character::scalar} for the first signature, and
+to \tcode{(BC).character} otherwise,
+where \tcode{BC} is the argument preceding the \tcode{character} in the signatures above.
+
+\pnum
+\tcode{is_kind} specifies \tcode{Q} to start a new hierarchy tree of a kind.
+
+\pnum
+Optional arguments may appear in any order.
+
+\pnum
+\begin{example}
+\begin{codeblock}
+// The first signature defines a base quantity.
+inline constexpr struct length final : quantity_spec<dim_length> {
+} length;  // Length is a base quantity.
+
+// The second signature defines a derived quantity.
+inline constexpr struct area final : quantity_spec<pow<2>(length)> {
+} area;  // An area equals length by length.
+
+// The third and fourth signatures add a leaf to a hierarchy of kinds.
+inline constexpr struct width final : quantity_spec<length> {
+} width;  // Width is a kind of length.
+
+// The fourth signature also refines the calculus required for implicit conversions.
+inline constexpr struct angular_measure final :
+    quantity_spec<dimensionless, arc_length / radius, is_kind> {
+} angular_measure;  // Requires an arc length per radius, not just any quantity of dimension one.
+\end{codeblock}
+\end{example}
+
+\begin{codeblock}
+namespace mp_units {
+
+template<@\defexposconceptnc{NamedQuantitySpec}@ Q>
+  requires requires { Q::dimension; }
+using @\exposidnc{to-dimension}@ = decltype(auto(Q::dimension));
+
+template<typename... Expr>
+struct @\exposidnc{derived-quantity-spec-impl}@ :
+    @\exposidnc{quantity-spec-interface}@,
+    @\exposidnc{expr-fractions}@<struct dimensionless, Expr...> {
+  using @\exposidnc{base-type}@ = @\exposidnc{derived-quantity-spec-impl}@;
+  using @\exposidnc{base}@ = @\exposidnc{expr-fractions}@<struct dimensionless, Expr...>;
+
+  static constexpr @\libconcept{Dimension}@ auto dimension =
+    @\exposidnc{expr-map}@<@\exposidnc{to-dimension}@, derived_dimension, struct dimension_one>(@\exposidnc{base}@{});
+  static constexpr quantity_character character =
+    @\exposidnc{derived-quantity-character}@(typename @\exposidnc{base}@::@\exposidnc{num}@{}, typename @\exposidnc{base}@::@\exposidnc{den}@{});
+};
+
+template<@\exposconceptnc{SymbolicConstant}@... Expr>
+struct @\libglobal{derived_quantity_spec}@ final : @\exposidnc{derived-quantity-spec-impl}@<Expr...> {};
+
+}
+\end{codeblock}
+
+\pnum
+\tcode{derived_quantity_spec} is used by the library
+to represent the result of a quantity calculus not equal to a named quantity.
+\begin{example}
+\begin{codeblock}
+auto area = pow<2>(length);
+static_assert(std::is_same_v<decltype(area), derived_quantity_spec<power<length, 2>>>);
+\end{codeblock}
+\end{example}
+A program that instantiates a specialization of \tcode{derived_quantity_spec}
+that is not a possible result of the library specifications
+is ill-formed, no diagnostic required.
+
+\begin{codeblock}
+namespace mp_units {
+
+struct @\libglobal{dimensionless}@ final : quantity_spec<derived_quantity_spec<>> {};
```

Comment:
If `dimensionless` is defined like this,
is there any point to `detail::derived_quantity_spec_impl`?
Its whole definition can be inlined into `derived_quantity_spec`.

Reply:
`derived_quantity_spec` is `final`.

Reply:
Yeah, but `dimensionless` derives from `quantity_spec<derived_quantity_spec<>>`.
I meant that `derived_quantity_spec` can inline the definition of `detail::derived_quantity_spec_impl`.

Reply:
Actually, it does not work :-(
A stupid clang with modules was compiled again, but the code was broken. This time, I checked other compilers as well before pushing changes.

The reason for it not compiling is that `kind_of` inherits from `Q::_base_type_` to inherit all of the properties of `Q`. If we merge those types, then `_base_type_` will be `final`. I could probably refactor `kind_of` to use composition and some mixins to clone the interface of `Q`, but the inheritance was the simplest option here.